exec/java-exec/src/main/java/org/apache/drill/exec/server/options/FallbackOptionManager.java

```diff
@@ -35,61 +42,65 @@ public FallbackOptionManager(OptionManager fallback) {
 
   @Override
   public Iterator<OptionValue> iterator() {
-    return Iterables.concat(fallback, optionIterable()).iterator();
+    return Iterables.concat(fallback, getLocalOptions()).iterator();
   }
 
   @Override
-  public OptionValue getOption(String name) {
-    final OptionValue opt = getLocalOption(name);
-    if(opt == null && fallback != null){
+  public OptionValue getOption(final String name) {
+    final OptionValue value = getLocalOption(name);
+    if (value == null && fallback != null) {
```

Comment:
This null check on the fallback manager seems odd to me. If something doesn't have a fallback, why would they create one of these? Should a non-null fallback be enforced in the constructor?